### ISSUE TITLE
Reverse slices result in negative size

### ISSUE THREAD

**Issue by bk-equityzen** [elastic/elasticsearch-dsl-py#1358]
```py
search[20:0]
```
results in `-20` as the size sent to ElasticSearch. This can possibly cause an error: `RequestError(400, 'search_phase_execution_exception', 'numHits must be > 0; please use TotalHitCountCollector if you just need the total hit count')`

Proposed Solution:
The slice function should result in a minimum size of `0`

```py
s._extra['size'] = n.stop - (n.start or 0) if n.stop is not None else 10
s._extra['size'] = max(s._extra['size'], 0)
```

**Comment by sethmlarson** [comment]
Nice catch! Would you be willing to provide a fix and test case for this issue?

**Comment by bk-equityzen** [comment]
> Nice catch! Would you be willing to provide a fix and test case for this issue?

Done! https://github.com/elastic/elasticsearch-dsl-py/pull/1360

**Comment by shanon-equityzen** [comment]
👋  @sethmlarson thanks for merging this in -- can you please let us know when we can expect this to be released?  🙌

**Comment by sethmlarson** [comment]
@shanon-equityzen I can try to get a patch release out tomorrow.

**Comment by bk-equityzen** [comment]
@sethmlarson any update on the next release?

**Comment by sethmlarson** [comment]
@bk-equityzen Released v7.2.1
